### PR TITLE
chore: fixes tests flakyness

### PR DESCRIPTION
**Pull request by nunomaduro**
This pull request applies the changes about the CTS concerting solving flakiness. Addresses https://github.com/algolia/algoliasearch-client-specs-internal/pull/105

### CODE REVIEW

File: tests/features/test_analytics_client.py [hunk 1]
```diff
@@ -10,17 +10,12 @@
...
-        self.index = F.index(self._testMethodName)
-        self.index2 = F.index('{}2'.format(self._testMethodName))
-
-    def tearDown(self):
-        self.index.delete()
-        self.index2.delete()
+        self.index = F.index(search_client, self._testMethodName)
+        self.index2 = F.index(search_client, '{}2'.format(self._testMethodName))  # noqa: E501
 
     def test_ab_testing(self):
```
Comment (nunomaduro) [inline tests/features/test_analytics_client.py]: In this test, i think we are going to have problems in the future saying `index don't exist`. The `add_abtest` should be retryable.

File: tests/helpers/factory.py [hunk 2]
```diff
@@ -21,8 +23,40 @@ def search_client(app_id=None, api_key=None):
...
+    def search_client_mcm():
+        # type: () -> SearchClient
+
+        app_id = os.environ['ALGOLIA_APPLICATION_ID_MCM']
+        api_key = os.environ['ALGOLIA_ADMIN_KEY_MCM']
+
+        return Factory.search_client(app_id, api_key)
+
+    @staticmethod
+    def index(search_client, index_name):
```
Comment (nunomaduro) [inline tests/helpers/factory.py]: For consistency, what do you about rename this method `search_index`?

Comment (aseure) [reply tests/helpers/factory.py]: As we don't have anything else called an index `index` and that `index` is well know term at Algolia, I'd keep this one if that's OK with you.

File: tests/features/test_analytics_client.py (hunk 1 above)
Comment (aseure) [reply tests/features/test_analytics_client.py]: Good catch indeed! Better to solve this now than later.

I've fixed this according to your recommendation and what you did for the JavaScript client: https://github.com/algolia/algoliasearch-client-python/pull/470/commits/a997d57edcadcb09448cc0fb5ed574c7d00e7c4c.

File: tests/features/test_search_client.py [hunk 3]
```diff
@@ -321,24 +321,36 @@ def test_dns_timeout(self):
...
             {
                 "validUntil": int(round(time.time())) + (60 * 10),  # + 10 min
-                "restrictIndices": self.index.name
+                "restrictIndices": index1.name
             }
         )
 
-        F.search_client(api_key=api_key).init_index(self.index.name).search('')
+        config2 = SearchConfig(F.get_app_id(), api_key)
+        config2.hosts = Factory.hosts(config2.app_id)
```
Comment (nunomaduro) [inline tests/features/test_search_client.py]:
```suggestion
        config2.hosts = config1.hosts
```

Shouldn't be the same hosts in all test?

Comment (aseure) [reply tests/features/test_search_client.py]: Yes definitely, stupid mistake. Fixed! Thanks buddy

**Pull request merged**